dev-packages/application-manager/src/generator/frontend-generator.ts

```diff
@@ -101,56 +101,71 @@ if (process.env.LC_ALL) {
 }
 process.env.LC_NUMERIC = 'C';
 
+const electron = require('electron');
 const { join } = require('path');
 const { isMaster } = require('cluster');
 const { fork } = require('child_process');
-const { app, BrowserWindow, ipcMain, Menu } = require('electron');
+const { app, BrowserWindow, ipcMain, Menu } = electron;
 
 const applicationName = \`${this.pck.props.frontend.config.applicationName}\`;
 const windows = [];
 
-function createNewWindow(theUrl) {
-    const newWindow = new BrowserWindow({ width: 1024, height: 728, show: !!theUrl, title: applicationName });
-    if (windows.length === 0) {
-        newWindow.webContents.on('new-window', (event, url, frameName, disposition, options) => {
-            // If the first electron window isn't visible, then all other new windows will remain invisible.
-            // https://github.com/electron/electron/issues/3751
-            options.show = true;
-            options.width = 1024;
-            options.height = 728;
-            options.title = applicationName;
-        });
-    }
-    windows.push(newWindow);
-    if (!!theUrl) {
-        newWindow.loadURL(theUrl);
-    } else {
-        newWindow.on('ready-to-show', () => newWindow.show());
-    }
-    newWindow.on('closed', () => {
-        const index = windows.indexOf(newWindow);
-        if (index !== -1) {
-            windows.splice(index, 1);
-        }
-        if (windows.length === 0) {
-            app.exit(0);
-        }
-    });
-    return newWindow;
-}
-
 if (isMaster) {
-    app.on('window-all-closed', () => {
-        if (process.platform !== 'darwin') {
-            app.quit();
-        }
-    });
-    ipcMain.on('create-new-window', (event, url) => {
-        createNewWindow(url);
-    });
     app.on('ready', () => {
         // Remove the default electron menus, waiting for the application to set its own.
         Menu.setApplicationMenu(Menu.buildFromTemplate([]));
+        const { screen } = electron;
+        const windows = [];
```

Comment:
One of the `const windows = [];` must go. Either this or [the other one](https://github.com/theia-ide/theia/pull/3479/files#diff-e6697ec7c0f7ddbbfd9e8365baacf132R111). +1 for removing the "other one".

Reply:
I think we noticed it at the same time :)

Fixed.